src/interceptor.ts

```diff
@@ -130,7 +130,8 @@ export const aws4Interceptor = <D = any>({
     }
 
     const { host, pathname, search } = new URL(url);
-    const { data, headers, method } = config;
+    const { data, method } = config;
+    const headers = new AxiosHeaders(config.headers);
```

Comment:
Maybe you should fix the actual headers when they are set at line 170:

https://github.com/jamesmbourne/aws4-axios/pull/1248/files#diff-bc2f1f907be8c49520a585dda7f0c100645e31e9bedf3828c3850fdf8389e619R170

```ts
config.headers = signingOptions.headers as AxiosHeaders;
```

This should not be casted but should be a real AxiosHeaders instance. Otherwise this lib will still break axios for every other uses outside this lib.